azurerm/internal/services/network/virtual_wan_data_source.go

```diff
@@ -0,0 +1,147 @@
+package network
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-05-01/network"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmVirtualWan() *schema.Resource {
```

Comment:
in order to confirm this Data Source works going forward (for example when we upgrade the Azure SDK/an API Version), we need to add an Acceptance Test to provision a Virtual WAN and confirm that this Data Source works as expected - as such could we add a test to check that? Here's an example of how we do that for `azurerm_subnet`: https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/network/tests/subnet_data_source_test.go

Reply:
done :)